cml-tdengine/src/core/inference.rs

```diff
@@ -149,6 +163,146 @@ mod tests {
+            .inherent_tag_num(1)
+            .optional_field_num(0)
+            .build()?;
+
+        fs::create_dir_all("/tmp/inference_dir/")?;
+        fs::write("/tmp/inference_dir/inference_data1.txt", b"8.8")?;
+        fs::write("/tmp/inference_dir/inference_data2.txt", b"9.8")?;
+        let mut batch_data_1 = vec![
+            NewSampleBuilder::default()
+                .data_path("/tmp/inference_dir/inference_data1.txt".into())
+                .build()?,
+            NewSampleBuilder::default()
+                .data_path("/tmp/inference_dir/inference_data2.txt".into())
+                .build()?,
+        ];
+
+        let available_status = vec!["SUCCESS"];
+        let last_batch_time: i64 = taos
+            .query_one(format!(
+                "SELECT LAST(ts) FROM task.`{}` where status IN ({}) ",
+                batch_meta_1.batch(),
+                available_status
+                    .iter()
+                    .map(|s| format!("'{}'", s))
+                    .collect::<Vec<String>>()
+                    .join(", ")
+            ))
+            .await?
+            .unwrap_or(0);
+        fs::write(
+            "/tmp/inference_dir/".to_string()
+                + batch_meta_1.batch()
+                + &last_batch_time.to_string()
+                + ".txt",
+            b"1",
+        )?;
+
+        let inference_fn = |vec_data: &mut Vec<NewSample<Value>>,
+                            batch: &str,
+                            task_time: i64|
+         -> Vec<NewSample<Value>> {
+            let mut result: Vec<NewSample<Value>> = Vec::new();
+            let working_dir = "/tmp/inference_dir/".to_string();
+            let model_inference =
+                fs::read_to_string(working_dir + batch + &task_time.to_string() + ".txt")
+                    .unwrap()
+                    .parse::<f32>()
+                    .unwrap();
+            for inference_data in vec_data.iter() {
+                // inference
+                let output = fs::read_to_string(inference_data.data_path())
+                    .unwrap()
+                    .parse::<f32>()
+                    .unwrap()
+                    + model_inference;
+                result.push(
+                    NewSampleBuilder::default()
+                        .data_path(inference_data.data_path().to_path_buf())
+                        .output(Some(Value::Float(output)))
+                        .build()
+                        .unwrap(),
+                );
+            }
+            result
+        };
+
+        tokio::spawn({
+            async move {
+                cml.inference(
+                    batch_meta_1,
+                    Field::new("output", Ty::Float, 8),
+                    available_status,
+                    &mut batch_data_1,
+                    &pool,
+                    inference_fn,
+                )
+                .await
+                .unwrap();
+            }
+        })
+        .await?;
```

Comment:
We may need more than one batch running simultaneously to validate if async works well.